src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -428,9 +434,16 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
 
     void addProjectionPart(const String & projection_name, std::shared_ptr<IMergeTreeDataPart> && projection_part);
 
+    void markProjectionPartAsBroken(const String & projection_name, const String & message, int code) const;
+
     bool hasProjection(const String & projection_name) const { return projection_parts.contains(projection_name); }
 
-    void loadProjections(bool require_columns_checksums, bool check_consistency, bool if_not_loaded = false);
+    bool hasBrokenProjection(const String & projection_name) const;
+
+    /// Return true, if all projections were loaded successfully and none was marked as broken.
```

Comment:
this comment doesn't make sense based on the current implementation
better yet, why don't we do what the comment is saying instead of using an extra variable for it?